src/java/org/apache/cassandra/utils/concurrent/NonBlockingRateLimiter.java

```diff
@@ -99,37 +99,61 @@ public int getRate()
     }
 
     /**
-     * Reserves a single permit which may not yet be available.
+     * Reserves a single permit slot on the timeline which may not yet be available.
      *
      * @return microseconds until the reserved permit will be available (or zero if it already is)
      */
     public long reserveAndGetWaitMicros()
```

Comment:
Might be nice to have the caller invoke with `MICROSECONDS` as a parameter? Very optional, just a suggestion,

Reply:
Sounds reasonable.